Develop

Melifaro2021 opened the pull request.

MaksBujanov committed:
Комментарии

MaksBujanov committed:
Комментарии

MaksBujanov committed:
MainActivity.kt реализовал двумя способами всплывающие сообщения
/* Способ 1. Лямбда-выражение */
/* Способ 2. Реализация анонимного класса  */
также попробовал работать с ImageView из кода.

MaksBujanov committed:
poster.png нужен был для временной работы с ImageView из кода

MaksBujanov committed:
делаю переходы на страницы

MaksBujanov committed:
Добавил два Toast 1. Реализация анонимного класса 2. Реализация через Лямбда-выражение

MaksBujanov committed:
Спринт 8/26 → Тема 5/6: Навигация → Урок 6/6
В обработчиках нажатия на кнопки нет Toast.
убраны.

MaksBujanov committed:
Шрифты

MaksBujanov committed:
Шрифты для MACOSX

MaksBujanov committed:
Подключил шрифты. убрал капс. изменил толщину линии начертания текста кнопок. добавил цветовые схемы

MaksBujanov committed:
Настроил светлую и темную темы.

ArturNurtdinov commented on app/src/main/AndroidManifest.xml:
```diff
@@ -10,17 +9,28 @@
-        android:theme="@style/Theme.Playlist_Maker"
+        android:theme="@style/Base.Theme.Playlist_Maker"
         tools:targetApi="31">
+        <!--android:theme="@style/Theme.Material3.DayNight.NoActionBar"-->
+        <!--android:theme="@style/Theme.Playlist_Maker" -->
+        <!--android:theme="@style/Theme.MaterialComponents.DayNight.NoActionBar" -->
```
Закоментированный код лучше сразу удалять. Хранить закоментированный код в проекте - плохая практика, в случае чего этот код можно восстановить с помощью истории гита

Melifaro2021 replied:
Хорошо, уберу.

ArturNurtdinov commented on app/src/main/java/ru/makskim/playlist_maker/MainActivity.kt:
```diff
@@ -1,11 +1,40 @@
+        // Кнопка Поиск
+        val searchButton = findViewById<Button>(R.id.search)
+        searchButton.setOnClickListener {
+            val displayIntent = Intent(
```
`displayIntent` лучше заинлайнить, либо дать нормальное название переменной, чтобы не было `displayIntent2` и `displayIntent3`

Melifaro2021 replied:
лучше заинлайнить, это что значит? Названия поменяю.

ArturNurtdinov commented on app/src/main/res/drawable/poster.png:
А этот файл зачем в проекте нужен?

Melifaro2021 replied:
Удалил. Первая проба заливки картинки.

ArturNurtdinov commented on app/src/main/res/font/._ys_display_medium.ttf:
Странно, что все файлы шрифтов называются с точки и нижнего подчеркивания. Даже студия на это ругается

Melifaro2021 replied:
Шрифты скопированы с архива выложенного в уроке яндекса. Видел , что многие ограничиваются копированием шрифтов только для андроида, а из папки для айфона не берут вообще. Я добавил для айфонов , но изменять оригинальные имена не стал, т.к. не знаю как это повлияет на работу со шрифтами. Если есть рекомендации , как изменить имена, слушаю.

ArturNurtdinov commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -1,164 +1,85 @@
-            >
+        android:background="@color/background_Settings_Page"
+        android:orientation="vertical"
+        tools:layout_editor_absoluteX="0dp"
+        tools:layout_editor_absoluteY="0dp">
```
Лучше убрать это, чтобы не мешало редактировать файл

Melifaro2021 replied:
Убрал

ArturNurtdinov commented on app/src/main/res/values/pageMain_style.xml:
```diff
@@ -0,0 +1,53 @@
+        <item name="android:paddingTop">10dp</item>
+        <item name="android:paddingRight">12dp</item>
+        <item name="android:paddingBottom">12dp</item>
+        <item name="android:paddingLeft">12dp</item>
```
Эти отступы нужно вынести в dimen.xml, они несколько раз используются

Melifaro2021 replied:
Внес.

ArturNurtdinov commented on app/src/main/res/values/pageSettings_style.xml:
```diff
@@ -0,0 +1,85 @@
+        <item name="android:paddingTop">4dp</item>
+        <item name="android:paddingRight">52dp</item>
+        <item name="android:paddingBottom">4dp</item>
+        <item name="android:paddingLeft">4dp</item>
```
А почему `paddingRight=52dp`? Получается несимметричная стрелка

ArturNurtdinov commented on app/src/main/res/values/pageSettings_style.xml:
```diff
@@ -0,0 +1,85 @@
+    <style name="pageSettings_string_head">
+        <item name="android:layout_width">match_parent</item>
+        <item name="android:layout_height">wrap_content</item>
+        <item name="android:color">@color/txt_Settings_Page</item>
```
`txt_Settings_Page` нужно назвать в соответствии со snake_case, то есть заглавных букв быть не должно

Melifaro2021 replied:
txt_Settings_Page .  camelCase знаю, там нет пробелов и спецсимволов, но заглавные обязательно присутствуют на стыке слов. snake_case ни разу не слышал такого понятия

Melifaro2021 replied:
txt_Settings_Page .  camelCase знаю, там нет пробелов и спецсимволов, но заглавные обязательно присутствуют на стыке слов. snake_case ни разу не слышал такого понятия

Melifaro2021 replied:
Переделал.

ArturNurtdinov commented on app/src/main/res/values/pageSettings_style.xml:
```diff
@@ -0,0 +1,85 @@
+        <item name="iconTint">@color/ico_Settings_Page</item>
+    </style>
+    <!-- Текст Заголовок настройки  -->
+    <style name="pageSettings_txt_settings">
```
Почему половина названия стиля в camelCase, а вторая половина - в snake_case? Нужно все стили называть в snake_case

Melifaro2021 replied:
Прочитал про snake_case Переделал названия по проекту по примеру: page_settings_txt_settings

ArturNurtdinov commented on app/src/main/res/values/pageSettings_style.xml:
```diff
@@ -0,0 +1,85 @@
+    </style>
+    <!-- Текст Заголовок настройки  -->
+    <style name="pageSettings_txt_settings">
+        <item name="android:text">match_parent</item>
```
Здесь точно не этот текст должен быть

MaksBujanov committed:
Сохранил в dimens.xml размеры :
1. Отступов
2. Шрифтов
3. Иконок

MaksBujanov committed:
1. изменил названия в соответствии со snake_case
2. Убрал poster.png
3. изменил названия для Intent
4. убрал закоментированный код и др

MaksBujanov committed:
Тест 1: Пробую перекрасить Svg иконки в зависимости от темы.
activity_settings.xml
android:src="@drawable/ico_arr_back"

MaksBujanov committed:
Перекрасил svg иконки для темной темы

MaksBujanov committed:
мелочи

MaksBujanov committed:
Обновил студию до версии Android Studio Giraffe | 2022.3.1 Patch 2
